src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSparkIntegrationTest.java

```diff
@@ -64,6 +65,7 @@ private String getResourceDir(){
                 //Note: these output files were created by running Picard 1.130 and GATK3.46
                 {new PipelineTest(GRCh37Ref2bit_chr2021, hiSeqBam_chr20, dbSNPb37_20, " --joinStrategy BROADCAST", getResourceDir() + EXPECTED_BAM)},
                 {new PipelineTest(GRCh37Ref_2021, hiSeqBam_chr20, dbSNPb37_20, " --joinStrategy SHUFFLE", getResourceDir() + EXPECTED_BAM)},
+                {new PipelineTest(GRCh37Ref2bit_chr2021, hiSeqBam_chr20_queryNameSorted, dbSNPb37_20, " --joinStrategy BROADCAST", getResourceDir() + EXPECTED_BAM)},
```

Comment:
Surprised that this works without https://github.com/HadoopGenomics/Hadoop-BAM/pull/49 -- I would have thought that the probability of getting a split between two reads in a pair would be fairly high given enough partitions...